koku/masu/external/downloader/azure/azure_service.py

```diff
@@ -173,14 +179,39 @@ def download_file(
             file_path = temp_file.name
 
         try:
-            blob_client = self._cloud_storage_account.get_blob_client(container_name, cost_export.name)
+            blob_client = self._cloud_storage_account.get_blob_client(container_name, key)
             with open(file_path, "wb") as blob_download:
                 blob_download.write(blob_client.download_blob().readall())
         except (AdalError, AzureException, ClientException, OSError) as error:
             raise AzureServiceError("Failed to download cost export. Error: ", str(error))
 
         return file_path
 
+    def get_blob(self, report, container_name):
```

Comment:
Could we use `get_file_for_key()` instead of adding another method? It's not as efficient since it's listing all the blobs but it's reusing existing code. (Side note: that method name is terrible and should probably be `get_properties_for_key`. Past me was so confused.).

Reply:
We totally can. Side note... :disappointed: get_file_for_key is also getting properties not a file/blob :stuck_out_tongue: .. Lets be honest naming is hard and MS makes it impossible when they have like 6 clients/objects for basically the same thing.